test/test_nn.py

```diff
@@ -15587,7 +15587,7 @@ def test_embedding_backward(self, device, dtype):
         self.assertEqual(embedding.weight.grad._indices(), tensorTwice)
         self.assertEqual(embedding.weight.grad._values(), onesTwice)
 
-    @dtypesIfCUDA(*ALL_TENSORTYPES2)
+    @dtypesIfCUDA(*get_all_fp_dtypes(include_bfloat16=TEST_WITH_ROCM))
```

Comment:
Do we want to keep using these `get_all_*` as mentioned in the previous comment.

cc: @mruberry @khushi-411

Reply:
Hi, @kshitij12345!
I stumbled upon the link provided by @mruberry. The macros it consists were:
```
- floating_types()
- floating_types_and_half()
- floating_types_and()
- floating_and_complex_types()
- floating_and_complex_types_and()
- integral_types()
- integral_types_and()
- all_types()
- all_types_and()
- complex_types()
- all_types_and_complex()
- all_types_and_complex_and()
- all_types_and_half()
```
None of these macros includes the `torch.bfloat16` type, that we require for testing with the `TEST_WITH_ROCM` environment.

Then I searched a couple of examples that use the `bfloat16` type with the `TEST_WITH_ROCM` environment and worked accordingly.

I think the other way without using `get_all_*` can be:
```
@dtypesIfCUDA(*(floating_types_and(dtypes=*(torch.half, if TEST_WITH_ROCM: torch.bfloat16)))
```
What do you think? Shall I replace it?

Thanks :)

Reply:
I think the `floating_types_and` would be preferable here

Reply:
Yes, I changed. Thanks :)

Reply:
Ideally we'd be rid of them but it's OK for now